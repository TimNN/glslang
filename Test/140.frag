#version 140

varying vec4 v;

in vec4 i;
out vec4 o;

void main()
{
}
