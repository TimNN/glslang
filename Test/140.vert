#version 140

uniform isamplerBuffer sbuf;

layout(std140) uniform blockName {
    int anonMem;
};

void main()
{
    int id = gl_InstanceID;
    id += anonMem;
    id += texelFetch(sbuf, 8).w;
    gl_ClipVertex;      // could be ERROR, but compiling under compatibility profile
    gl_Color;           // could be ERROR, but compiling under compatibility profile
    gl_LightSource[0];  // could be ERROR, but compiling under compatibility profile
    gl_DepthRange.far;
    gl_TexCoord;        // could be ERROR, but compiling under compatibility profile
    gl_FogFragCoord;    // could be ERROR, but compiling under compatibility profile
    gl_FrontColor;      // could be ERROR, but compiling under compatibility profile
}

out vec4 gl_Position;  // ERROR

layout(location = 9) in vec4 locBad;  // ERROR

#extension GL_ARB_explicit_attrib_location : enable

layout(location = 9) in vec4 loc;

#extension GL_ARB_separate_shader_objects : enable

out vec4 gl_Position;
in vec4 gl_Position;   // ERROR
out vec3 gl_Position;  // ERROR

out float gl_PointSize;
out vec4 gl_ClipVertex;
out float gl_FogFragCoord;
