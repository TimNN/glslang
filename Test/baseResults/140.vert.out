140.vert
ERROR: 0:23: 'gl_Position' : identifiers starting with "gl_" are reserved 
ERROR: 0:28: 'redeclaration' : cannot change storage, memory, or auxiliary qualification of gl_Position
ERROR: 0:28: 'redeclaration' : cannot change interpolation qualification of gl_Position
ERROR: 0:29: 'redeclaration' : cannot change the type of gl_Position
ERROR: 0:32: 'gl_ClipVertex' : cannot redeclare after use 
ERROR: 0:33: 'gl_FogFragCoord' : cannot redeclare after use 
ERROR: 6 compilation errors.  No code generated.


Shader version: 140
Requested GL_ARB_separate_shader_objects
ERROR: node is still EOpNull!
0:9  Function Definition: main( (void)
0:9    Function Parameters: 
0:11    Sequence
0:11      Sequence
0:11        move second child to first child (int)
0:11          'id' (int)
0:11          'gl_InstanceID' (gl_InstanceId int)
0:12      add second child into first child (int)
0:12        'id' (int)
0:12        anonMem: direct index for structure (layout(column_major std140 offset=0 ) uniform int)
0:12          'anon@0' (layout(column_major std140 ) uniform block{layout(column_major std140 offset=0 ) uniform int anonMem})
0:12          Constant:
0:12            0 (const uint)
0:13      add second child into first child (int)
0:13        'id' (int)
0:13        direct index (int)
0:13          Function Call: texelFetch(isB1;i1; (4-component vector of int)
0:13            'sbuf' (uniform isamplerBuffer)
0:13            Constant:
0:13              8 (const int)
0:13          Constant:
0:13            3 (const int)
0:14      'gl_ClipVertex' (gl_ClipVertex 4-component vector of float)
0:15      'gl_Color' (in 4-component vector of float)
0:16      direct index (structure{4-component vector of float ambient, 4-component vector of float diffuse, 4-component vector of float specular, 4-component vector of float position, 4-component vector of float halfVector, 3-component vector of float spotDirection, float spotExponent, float spotCutoff, float spotCosCutoff, float constantAttenuation, float linearAttenuation, float quadraticAttenuation})
0:16        'gl_LightSource' (uniform 32-element array of structure{4-component vector of float ambient, 4-component vector of float diffuse, 4-component vector of float specular, 4-component vector of float position, 4-component vector of float halfVector, 3-component vector of float spotDirection, float spotExponent, float spotCutoff, float spotCosCutoff, float constantAttenuation, float linearAttenuation, float quadraticAttenuation})
0:16        Constant:
0:16          0 (const int)
0:17      far: direct index for structure (float)
0:17        'gl_DepthRange' (uniform structure{float near, float far, float diff})
0:17        Constant:
0:17          1 (const int)
0:18      'gl_TexCoord' (smooth out implicitly-sized array of 4-component vector of float)
0:19      'gl_FogFragCoord' (smooth out float)
0:20      'gl_FrontColor' (smooth out 4-component vector of float)
0:?   Linker Objects
0:?     'sbuf' (uniform isamplerBuffer)
0:?     'anon@0' (layout(column_major std140 ) uniform block{layout(column_major std140 offset=0 ) uniform int anonMem})
0:?     'gl_TexCoord' (smooth out implicitly-sized array of 4-component vector of float)
0:?     'gl_Position' (smooth out 4-component vector of float)
0:?     'gl_PointSize' (gl_PointSize float)
0:?     'gl_PointSize' (gl_PointSize float)
0:?     'gl_ClipVertex' (gl_ClipVertex 4-component vector of float)
0:?     'gl_ClipVertex' (gl_ClipVertex 4-component vector of float)
0:?     'gl_FogFragCoord' (smooth out float)
0:?     'gl_FogFragCoord' (smooth out float)
0:?     'gl_VertexID' (gl_VertexId int)
0:?     'gl_InstanceID' (gl_InstanceId int)


Linked vertex stage:


Shader version: 140
Requested GL_ARB_separate_shader_objects
ERROR: node is still EOpNull!
0:9  Function Definition: main( (void)
0:9    Function Parameters: 
0:11    Sequence
0:11      Sequence
0:11        move second child to first child (int)
0:11          'id' (int)
0:11          'gl_InstanceID' (gl_InstanceId int)
0:12      add second child into first child (int)
0:12        'id' (int)
0:12        anonMem: direct index for structure (layout(column_major std140 offset=0 ) uniform int)
0:12          'anon@0' (layout(column_major std140 ) uniform block{layout(column_major std140 offset=0 ) uniform int anonMem})
0:12          Constant:
0:12            0 (const uint)
0:13      add second child into first child (int)
0:13        'id' (int)
0:13        direct index (int)
0:13          Function Call: texelFetch(isB1;i1; (4-component vector of int)
0:13            'sbuf' (uniform isamplerBuffer)
0:13            Constant:
0:13              8 (const int)
0:13          Constant:
0:13            3 (const int)
0:14      'gl_ClipVertex' (gl_ClipVertex 4-component vector of float)
0:15      'gl_Color' (in 4-component vector of float)
0:16      direct index (structure{4-component vector of float ambient, 4-component vector of float diffuse, 4-component vector of float specular, 4-component vector of float position, 4-component vector of float halfVector, 3-component vector of float spotDirection, float spotExponent, float spotCutoff, float spotCosCutoff, float constantAttenuation, float linearAttenuation, float quadraticAttenuation})
0:16        'gl_LightSource' (uniform 32-element array of structure{4-component vector of float ambient, 4-component vector of float diffuse, 4-component vector of float specular, 4-component vector of float position, 4-component vector of float halfVector, 3-component vector of float spotDirection, float spotExponent, float spotCutoff, float spotCosCutoff, float constantAttenuation, float linearAttenuation, float quadraticAttenuation})
0:16        Constant:
0:16          0 (const int)
0:17      far: direct index for structure (float)
0:17        'gl_DepthRange' (uniform structure{float near, float far, float diff})
0:17        Constant:
0:17          1 (const int)
0:18      'gl_TexCoord' (smooth out 1-element array of 4-component vector of float)
0:19      'gl_FogFragCoord' (smooth out float)
0:20      'gl_FrontColor' (smooth out 4-component vector of float)
0:?   Linker Objects
0:?     'sbuf' (uniform isamplerBuffer)
0:?     'anon@0' (layout(column_major std140 ) uniform block{layout(column_major std140 offset=0 ) uniform int anonMem})
0:?     'gl_TexCoord' (smooth out 1-element array of 4-component vector of float)
0:?     'gl_Position' (smooth out 4-component vector of float)
0:?     'gl_PointSize' (gl_PointSize float)
0:?     'gl_PointSize' (gl_PointSize float)
0:?     'gl_ClipVertex' (gl_ClipVertex 4-component vector of float)
0:?     'gl_ClipVertex' (gl_ClipVertex 4-component vector of float)
0:?     'gl_FogFragCoord' (smooth out float)
0:?     'gl_FogFragCoord' (smooth out float)
0:?     'gl_VertexID' (gl_VertexId int)
0:?     'gl_InstanceID' (gl_InstanceId int)

